### PR TITLE
Refactor FXIOS-9052 - Add a11y identifier to BottomSheetViewModel

### PR DESCRIPTION
**Pull request by DanielDervishi**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-9052)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/20019)

## :bulb: Description

@thatswinnie I understand that you pointed out the lack of a11y identifiers associated with BottomSheetViewModel, so Cyndi and I thought it would be best to ping you as well!

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: BrowserKit/Sources/ComponentLibrary/BottomSheet/BottomSheetViewModel.swift [hunk 1]
```diff
@@ -19,27 +19,31 @@ public struct BottomSheetViewModel {
     public var shouldDismissForTapOutside: Bool
     public var shadowOpacity: Float
     public var closeButtonA11yLabel: String
+    public var closeButtonA11yIdentifier: String
 
-    public init(closeButtonA11yLabel: String) {
+    public init(closeButtonA11yLabel: String, closeButtonA11yIdentifer: String) {
```
Comment (cyndichin) [inline BrowserKit/Sources/ComponentLibrary/BottomSheet/BottomSheetViewModel.swift:24]:
```suggestion
    public init(closeButtonA11yLabel: String, closeButtonA11yIdentifier: String) {
```

**Review by cyndichin — APPROVED** [review]
great job @DanielDervishi !! looks great, one small nit on spelling I just noticed. Also, I added @thatswinnie as a reviewer to get her eyes on this as well (don't think you have access to do that yet).

**Review by cyndichin — APPROVED** [review]
🔥

File: firefox-ios/Client/Coordinators/AddressAutofillCoordinator.swift [hunk 2]
```diff
@@ -44,7 +44,7 @@ class AddressAutofillCoordinator: BaseCoordinator {
     /// Shows the address autofill functionality within a bottom sheet.
     /// - Parameter frame: The WKFrameInfo object representing the web view frame.
     func showAddressAutofill(frame: WKFrameInfo?) {
-        let bottomSheetViewModel = BottomSheetViewModel(closeButtonA11yLabel: .CloseButtonTitle)
+        let bottomSheetViewModel = BottomSheetViewModel(closeButtonA11yLabel: .CloseButtonTitle, closeButtonA11yIdentifier: AccessibilityIdentifiers.Autofill.addressCloseButton)
```
Comment (thatswinnie) [inline firefox-ios/Client/Coordinators/AddressAutofillCoordinator.swift:47]: This line will cause a Swiftlint warning as it is too long. The same will happen for all the other places you add `closeButtonA11yIdentifier` to the initializer of `BottomSheetViewModel`.

**Review by thatswinnie — COMMENTED** [review]
Looks great apart from the Swiftlint line length warnings. Good job @DanielDervishi 🎉

**Review by thatswinnie — APPROVED** [review]

**Comment by cyndichin** [comment]
awesome work @DanielDervishi ! thank you!

**Pull request merged**